### ISSUE TITLE
Mastodon probes invalid URL on ActivityPub follow

### ISSUE THREAD

**Issue by zotlabs** [tootsuite/mastodon#4932]
First brought up in #4906 but moved to separate issue.

TL;DR Mastodon appears to attempt Follow validation on a URL which was not provided by us and this fails. Us is the red/hubzilla ActivityPub service.

A Follow activity was sent to Mastodon which looks like this:

````
{
  "@context":[
    "https://www.w3.org/ns/activitystreams",
    "https://w3id.org/security/v1",
    "https://hz.macgirvin.com/apschema"
  ],
  "id":"https://hz.macgirvin.com/follow/190",
  "type":"Follow",
  "actor":{
    "type":"Person",
    "id":"https://hz.macgirvin.com/channel/grek",
    "preferredUsername":"grek",
    "name":"grek",
    "icon":{
      "type":"Image",
      "mediaType":"image/png",
      "url":"https://hz.macgirvin.com/photo/profile/l/15",
      "height":300,
      "width":300
    },
    "url":{
      "type":"Link",
      "mediaType":"text/html",
      "href":"https://hz.macgirvin.com/channel/grek"
    },
    "inbox":"https://hz.macgirvin.com/inbox/grek",
    "outbox":"https://hz.macgirvin.com/outbox/grek",
    "endpoints":{
      "sharedInbox":"https://hz.macgirvin.com/inbox/[public]"
    },
    "publicKey":{
      "id":"https://hz.macgirvin.com/channel/grek/public_key_pem",
      "owner":"https://hz.macgirvin.com/channel/grek",
      "publicKeyPem":"-----BEGIN PUBLIC KEY-----
MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAyu/X+8WTr42z+P87Q44v
Q9n7zR9wlXxEe4xliDo0U3Lm0JEIxq3Rdip+0G+5FE6/SmJ0EJqPDyNRiCQ2mq86
OOOohzJi+nTswH8S2xoNO0cgWW5ioXq5kohtBRvSrdyMBFlZOfKVwV59Mug8iyI9
WT+NqgBREkS/aygdu/pdHhOW9S1lxfwhvDFuK037j4dUGgcaxveRhu9dTDXXiYQ7
gwviXc88P5pqPCHVV1jVwI7AaOe3IyuHPzAzwJxlTISa/7s4eCqK9oaFQZLGv1SJ
+YkRdWSHuY2g3y/gHR6plRvv4F9g8P3GaV7k0okL5Knd/XrVi68eU7KSZV7kXR7D
yVwXwR4KbPdFKDNd3jEFdMgzlaNo7Gw/ezZ5/SJUxkelPMrSrRLH9sZPU/aMFC3G
QKJrW7iu0SVDYcYLauR7z4TNu2nNfxxUrW//HFxt9m6dOcLiZg6PaPIYQV8ABWG7
Mx9d28Fd+nnEfHBno9+V19rb86qzgZI+Xf5bh+tCAZ5G7jbdCIpzHIto3NEl+Dka
XtCM6my+x2yEXLYJfyzIVYWt+yE3EdCzolycE+Re6CCgTTP7cTEJxwCzt8B/B5cT
LCIcTOnwMkuZa4nN0+Wwd6Gm4a7jmT/NPCfDvkJTs5YYH4lTjcjnkeD2Ac+fa1Ie
hgFi/S9/tnSQP7zS2Cpohp0CAwEAAQ==
-----END PUBLIC KEY-----
"
    },
    "nomadicLocations":[
      {
        "id":"https://hz.macgirvin.com/locs/grek",
        "type":"nomadicLocation",
        "locationAddress":"acct:grek@hz.macgirvin.com",
        "locationPrimary":true,
        "locationDeleted":false
      },
      {
        "id":"https://z2.macgirvin.com/locs/grek",
        "type":"nomadicLocation",
        "locationAddress":"acct:grek@z2.macgirvin.com",
        "locationPrimary":false,
        "locationDeleted":false
      }
    ]
  },
  "object":{
    "type":"Person",
    "id":"https://mastodon.social/users/hzplanet",
    "name":"hzplanet",
    "icon":{
      "type":"Image",
      "mediaType":"image/png",
      "url":"https://hz.macgirvin.com/photo/789deb5536a325ece2885d562a3b2fc0-4",
      "height":300,
      "width":300
    },
    "url":{
      "type":"Link",
      "mediaType":"text/html",
      "href":"https://mastodon.social/users/hzplanet"
    },
    "inbox":"https://hz.macgirvin.com/nullbox",
    "outbox":"https://hz.macgirvin.com/nullbox"
  },
  "signature":{
    "@context":[
      "https://www.w3.org/ns/activitystreams",
      "https://w3id.org/security/v1"
    ],
    "type":"RsaSignature2017",
    "nonce":"48fac7028155b6811022b0faf9e5d4ad452dec1be5aeb7952b183976c7b4d28c",
    "creator":"https://hz.macgirvin.com/channel/grek/public_key_pem",
    "created":"2017-09-14T12:27:14Z",
    "signatureValue":"NIbBqJNJm8fHlqfH7gZOApmmYTm6uyJWz8t2QVuqevweCvDp9nIU7jhpV+NsyHaGQCcAek79I+ImEd1euJFG8RA3dbaIIYsMfmTrl5WWrLJwm7ppId4kVp0YLOnRftGoY88u4qTnGyJm3MbO9FL34ZeRgEg4WQzx87Ebw5/IrPav/peDKoUm/biXoJQm6U2xXExKmgKQbbRcqVbgCswKNpgm51YQ9w9AMG137P+XMluZ+gaWUV+LoHH/4zUJ+c1VR/V5ABgGU+a/bSbxyqBqOfIoZxCG6yTjfbyZkkjhYGbEfDfE641L67Kv8OQ+gF6AvtReMfMozHbUgZG6KYs7mAbRxJc6DF2ub5KCL1QHKY+b7GBbWB8Bd99HJk1zXk9/s0ftZeRXrniE5ToKMvKv+yY4ibn4alEX/V6xQxED39rpSNppFe+GjbvB5wrFjATLNXQU18ZwGBZk56Gsxcj7famRFdxhpTmFIddjL2Kj5X/2UCqYvKmiqkBP2ZiwNVqOvQKq6lmeSYr0Nz0+GscnLBi79lj8Yu9pdZbdqN03qQDQQ2e7GPyIxWDuEynGHSetQpdvOK31fp4ySSYO7cgRiiot8mMFiiWww9117MJWbU9BG0HvxW5is7egozhGB8Uh16jonPW5SSaBgf1Xu0KVJhDTKlh4DEC2+aWYEPEVRrk="
  }
}
````

We're aware of the bogus inbox on the object actor and will fix that. Was trying to come up with a way to make the object spec compliant. 

Anyway the issue is that the Follow activity has no effect; however the server in the object does make a GET request to 

https://hz.macgirivin.com/channel

This URL is not present in the follow activity and the request is denied (404). *If* we direct the failed URL to the actual actor URL, the Follow succeeds, but we cannot do this in production since the part of the URL that is missing is the actual actor involved in the transaction 

We supply HTTP Signatures with a keyId in this case of "https://hz.macgirvin.com/channel/grek". The URL can also be probed via webfinger but there is no evidence Mastodon made a webfinger request during the failed sequence. 
 
* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate (mentioned in #4906).
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by zotlabs** [comment]
I tracked this to an issue in our code. The bogus url is coming from the http signature keyId; which was incorrectly set by our queue manager. It's correct when viewing the activities by URI but was incorrectly generated for queued outbound deliveries.
